### PR TITLE
Restore forgotten passwords

### PR DESCRIPTION
**Pull request by dostuffthatmatters**

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -73,3 +74,18 @@ async def change_password_route(
 ):
     account = await authenticate_from_token(access_token)
     return await change_password(account, old_password, new_password)
+
+
+@app.post('/forgot-password')
```
Comment (empicano) [inline app/routes.py:79]: Our hero [Florimond](https://florimond.dev/blog/articles/2018/08/restful-api-design-13-best-practices-to-make-your-users-happy/) recommends not using verbs as REST routes (same comment for most of the routes)

File: app/utilities/account_functions.py [hunk 2]
```diff
@@ -6,7 +6,8 @@
 
 from app.utilities.encryption import generate_password_hash, \
     check_password_hash, generate_secret_token, validate_password_format
-from app.utilities.mailing import send_verification_mail
+from app.utilities.mailing import \
+    send_verification_mail, send_forgot_password_mail
 
 
 async def get_account(email: str):
```
Comment (empicano) [inline app/utilities/account_functions.py:13]: If you don't want all the data in the database, better use MongoDB's `filter` keyword with `find_one` -- that's more performant as you'll simply not load those fields from the database.

File: app/utilities/mailing.py [hunk 3]
```diff
@@ -6,22 +6,48 @@ async def send_verification_mail(db_account):
     data = {
         'from': 'FastSurvey <noreply@admin.fastsurvey.io>',
```
Comment (empicano) [inline app/utilities/mailing.py]: I would like to stick with `noreply@fastsurvey.io` -- less email addresses are easier to manage

File: app/utilities/mailing.py [hunk 4]
```diff
@@ -6,22 +6,48 @@ async def send_verification_mail(db_account):
...
     verification_url = (
-        ADMIN_FRONTEND_URL + "/verification?token=" + db_account["email_token"]
+        ADMIN_FRONTEND_URL + "/verify-email?token=" + db_account["email_token"]
     )
     return (
         '<h1>Welcome to FastSurvey!</h1>' +
-        '<p>Please verify this email vie the following link:</p>' +
+        '<p>Please verify this email via the following link:</p>' +
+        f'<a href="{verification_url}" target="_blank">{verification_url}</a>' +
+        '<p>Best,<br/>The FastSurvey Team</p>' +
```
Comment (empicano) [inline app/utilities/mailing.py]: I think the correct spelling is: _Best, the FastSurvey team_

File: app/utilities/mailing.py [hunk 5]
```diff
@@ -6,22 +6,48 @@ async def send_verification_mail(db_account):
...
+        '<p><br/>If you have not signed up for this ' +
+        'service, you can just ignore this email.</p>'
+    )
+
+
+async def send_forgot_password_mail(email: str, token: str):
+    data = {
+        'from': 'FastSurvey <noreply@admin.fastsurvey.io>',
```
Comment (empicano) [inline app/utilities/mailing.py]: Same here as above

**Comment by empicano** [comment]
Our password for all our services is `password123456789101112131415`, please write it down so you don't forget again.

File: app/routes.py (hunk 1 above)
Comment (dostuffthatmatters) [reply app/routes.py:79]: This backend is not restful 😅😅 Maybe I will make it more REST compliant some time.

File: app/utilities/account_functions.py (hunk 2 above)
Comment (dostuffthatmatters) [reply app/utilities/account_functions.py:13]: Yes! thank you :)

File: app/utilities/mailing.py (hunk 3 above)
Comment (dostuffthatmatters) [reply app/utilities/mailing.py]: Will do!

File: app/utilities/mailing.py (hunk 4 above)
Comment (dostuffthatmatters) [reply app/utilities/mailing.py]: Yes, that is better!

**Comment by dostuffthatmatters** [comment]
@empicano Thank you for the helpful hints! :)

**Pull request merged**